lib/src/Uncertainty/Bayesian/openturns/MetropolisHastings.hxx

```diff
@@ -0,0 +1,97 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Abstract top-level class for all MetropolisHastings
+ *
+ *  Copyright 2005-2021 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_METROPOLISHASTINGS_HXX
+#define OPENTURNS_METROPOLISHASTINGS_HXX
+
+#include "openturns/TypedInterfaceObject.hxx"
+#include "openturns/MetropolisHastingsImplementation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class MetropolisHastings
+ *
+ * @brief The class describes the concept of a calibration strategy.
+ *
+ *
+ * The implementation defines how the
+ * @see MetropolisHastingsImplementation
+ */
+class OT_API MetropolisHastings
+  : public TypedInterfaceObject<MetropolisHastingsImplementation>
+{
+  CLASSNAME
+public:
+
+  /** Default constructor */
+  MetropolisHastings();
+
+  /** Constructor from implementation */
+  MetropolisHastings(const MetropolisHastingsImplementation & implementation);
+
+#ifndef SWIG
+  /** Constructor from implementation pointer */
+  MetropolisHastings(MetropolisHastingsImplementation * p_implementation);
+#endif
+
+  /** Get a realization */
+  Point getRealization() const;
+  
+  /** Prior accessor */
+  void setTargetDistribution(const Distribution & prior);
+  Distribution getTargetDistribution() const;
+
+  /** Initial state accessor */
+  Point getInitialState() const;
+
+  /** Burnin accessor */
+  void setBurnIn(UnsignedInteger burnIn);
+  UnsignedInteger getBurnIn() const;
+
+  /** Thinning accessor */
+  void setThinning(UnsignedInteger thinning);
+  UnsignedInteger getThinning() const;
+
+  /** History accessor */
+  void setHistory(const HistoryStrategy & strategy);
+  HistoryStrategy getHistory() const;
+  
+  /** Dimension accessor */
+  UnsignedInteger getDimension() const;
+
+  /** Likelihood accessor */
+  void setLikelihood(const Distribution & conditional,
+                     const Sample & observations,
+                     const Function & model = Function(),
+                     const Sample & covariates = Sample(0, 0));
+
+  /** Compute the log-likelihood */
+  Scalar computeLogPosterior(const Point & currentState) const;
```

Comment:
The actual `computeLogLikelihood` method is missing.

Other methods are also missing:
- getConditional
- getCovariates
- getLinkFunction
- getObservations
- getTargetLogPDF
- getTargetLogPDFSupport

I also think we should either remove setTargetDistribution or add setTargetLogPDF/setTargetLogPDFSupport.

If we add setTargetLogPDF/setTargetLogPDFSupport, we need to handle user mistakes like defining a real OT Distribution as target but then calling setTargetLogPDFSupport.

Reply:
Also, there are still a few mentions of "MetropolisHastingsImplementation" in the doc.

Reply:
yes